### ISSUE TITLE
Define 4.15.0 release criteria

### ISSUE THREAD

**Issue by abought** [statgen/raremetal#5]
## Goal
We have merged #2 (SKAT bugfixes) to the `develop` branch, alongside a few unit tests (based on the public tutorial as a baseline). These features have not been released/announced to the general public.

The output of all tests now better matches v4.13.9 (before the known SKAT issues were introduced).  This does not imply that either the previous or current versions are inherently accurate, however. We should identify "good enough" criteria that allow a release, and set procedures for spot-checking future releases.

A list of known issues for triage is below, to identify if any of them constitute total blockers.

## Current validation status
- The RAREMETAL tutorial (1k SNPs, 742 individuals) has been used as the base dataset. Unit tests compare output to that expected for 4.13.9.
- Copies of the `develop` have been sent out to several researchers for feedback but we have not heard back yet. We may need to run on a larger dataset ourselves.
- Annotations/ deeper review in progress

## Release notes so far
- Fix known/reported issues that cause incorrect results to be output from SKAT tests
- Changes to file I/O:  port uses of libStatGen I/O to use Savvy 
  - The new file reader library does not support RAREMETALWORKER's `noeof` and `flagDosage` options, and these command line flags have been removed. 
  - Most well-formatted files should not require these options. Some older files may need to be re-indexed with tabix.
- Convert build process to use CMake and improve cross-platform compatibility. See README.md for information on how to compile.
- Add tutorial data to the repository

### May impact release
- Need to establish licensing (GPL? Other?) Will need approval of all known contributors.
- Unit tests are diff-based and do not pass on all platforms (due to platform-specific floating point differences in least significant digits)
- ~~Possible regression: as of savvy port, `flagDosage` CLI option [may have no effect](https://github.com/statgen/raremetal/commit/c2bab997d31792e1c0764d3f6cf33b673404e9c8#diff-d6245426b9d373477a7e6c93e71a4e4bL109)~~
- ~~Clarify intended behavior: `noeof` option [not supported](https://github.com/abought/raremetal/blob/a56400ca037ace1546e6c69eb66a77f85bb8c31d/raremetalworker/src/Main.cpp#L123) in savvy~~

### Possible future work
- Investigating uncertain covariance matrix behavior in Raremetalworker (Laura)
- Argument parsing (#3): program fails to stop on missing or ambiguous options. 
- Various truncation and type conversion issues (#4)
- [Extreme](https://github.com/abought/raremetal/blob/a56400ca037ace1546e6c69eb66a77f85bb8c31d/raremetal/src/Meta.cpp#L2432) pvalues are handled in a way that may mis-estimate actual significance (various occurrences throughout code)

**Comment by welchr** [comment]
Must fix: 

* Every scenario in which a variant is filtered out should be logged. There are currently a few places where this does not happen. 

* Fix type conversion bug when estimating genotype counts under the `--dosages` option (#4)

* Fix using libStatGen parameter parser for ambiguous parameters (#3), or alternatively the program should halt when an ambiguous parameter is detected. 

Possible future work: 

* Why does the program use N and ALT AF to determine genotype counts when using `--dosages`, and then use those counts to determine whether the variant is monomorphic, instead of directly using ALT AC in the file?

**Comment by abought** [comment]
We identified an issue with a newer commit on the develop branch. Having restored this branch to a state of "working unit tests that match 4.13.9", we've pushed out release [4.15.0](https://github.com/statgen/raremetal/releases/tag/v4.15.0) that contains these bugfixes. (cc: @abecasis , @gjmzajac )

Scope and impacts:
- For available unit tests, this new release should restore RAREMETAL to matching output of 4.13.9 for all available tutorial examples.
- All 4.14.x releases are affected by a bug that impacts results for SKAT and burden test methods
- Any users of the `develop` branch *may* be affected, as follows:
  - Work based on the March 22 `develop` branch should be ok
  - Work based on the April 19 `develop` branch (a1e648edc98a1097e79f18ecf4f2ffd69542ba1b) may be subject to a bug that affected which variants are read for analysis. This problem could materially alter calculated results.
  - The bug has been fixed in the official 4.15.0 release linked above, and we have cut a release so that people can have an "official" reference point for analysis.
